x/multisig/decorator.go

```diff
@@ -22,47 +22,59 @@ func NewDecorator() Decorator {
 func (d Decorator) Check(ctx weave.Context, store weave.KVStore, tx weave.Tx,
 	next weave.Checker) (weave.CheckResult, error) {
 	var res weave.CheckResult
-	err := d.withMultisig(ctx, store, tx)
+	newCtx, err := d.withMultisig(ctx, store, tx)
 	if err != nil {
 		return res, err
 	}
 
-	return next.Check(ctx, store, tx)
+	return next.Check(newCtx, store, tx)
 }
 
 // Deliver enforces multisig contract before calling down the stack
 func (d Decorator) Deliver(ctx weave.Context, store weave.KVStore, tx weave.Tx,
 	next weave.Deliverer) (weave.DeliverResult, error) {
 	var res weave.DeliverResult
-	err := d.withMultisig(ctx, store, tx)
+	newCtx, err := d.withMultisig(ctx, store, tx)
 	if err != nil {
 		return res, err
 	}
 
-	return next.Deliver(ctx, store, tx)
+	return next.Deliver(newCtx, store, tx)
 }
 
-func (d Decorator) withMultisig(ctx weave.Context, store weave.KVStore, tx weave.Tx) error {
+func (d Decorator) withMultisig(ctx weave.Context, store weave.KVStore, tx weave.Tx) (weave.Context, error) {
 	if multisigContract, ok := tx.(MultiSigTx); ok {
 		// does tx have multisig ?
 		addr := multisigContract.GetMultiSig()
 		if addr == nil {
-			return nil
+			return ctx, nil
+		}
+
+		if d.auth.HasAddress(ctx, addr) {
```

Comment:
Confusion due to poor wording.

You call it `addr`.
Here you check if it is an Address in auth.
5 lines below, you use it as a primary key to load a model from a bucket.

it is addr or pk. I think the correct one is `pk`